cpp/src/arrow/io/memory.h

```diff
@@ -40,6 +40,17 @@ class ARROW_EXPORT BufferOutputStream : public OutputStream {
  public:
   explicit BufferOutputStream(const std::shared_ptr<ResizableBuffer>& buffer);
 
+  /// \brief Create in-memory output stream with a default capacity of 1K using
```

Comment:
Hmm... I'm not sure why this API is useful? Why 1K specifically?

Reply:
It's more convenient than writing `BufferOutputStream::Create(1024, default_memory_pool(), &stream)`. It's nice to have a simple default along with a more customizable version. I can remove it, too.

1K is arbitrary. It could easily be 4K

Reply:
FWIW, 1K seems to be a popular number for initializing in-memory buffers

* https://github.com/apache/impala/blob/64e6719870db5602a6fa85014bc6c264080b9414/be/src/rpc/thrift-util.h#L48

* https://github.com/apache/arrow/blob/master/cpp/src/parquet/util/memory.h#L50

Reply:
I see. Sounds fine to me then.